Content Comments

ssddanbrown opened the pull request:
Related to #1265

## Todo

- [x] Inline comment display
  - [ ] ~~Option for controlling display~~
    - Omitting this for initial implementation as we'll gather some feedback about best placement and control level.
  - [x] Selection zone representation (on focus/click?)
  - [x] Indicator for if the comment is outdated
- [x] WYSIWYG display?
- [x] Update reference hash to be focused on text content, otherwise too much variance (eg, when loaded in editor).
- [ ] ~~MD editor display?~~
  - The MD editor cycle is too loose to retain accurate references, so retaining editor support to the new WYSIWYG.
- [x] Comment archival handling
- [x] Test changes to pointer work across presentation variation (change of edit/comment permissions etc...)
- [x] Test accessible use of pointer remains, including added comment action
- [x] Test use in old WYSIWYG (references remain inactive but don't break anything).

ssddanbrown committed:
Comments: Started logic for content references

Adds button for comments to pointer.
Adds logic to generate a content reference point.

ssddanbrown committed:
Comments & Pointer: Converted components to typescript

Made changes for dom and translation services for easier usage
considering types.
trans_choice updated to allow default count replacement data as per
Laravel's default behaviour.

ssddanbrown committed:
Comments: Added back-end content reference handling

Also added archived property, to be added.

ssddanbrown committed:
Comments: Converted comment component to TS

ssddanbrown committed:
Comments: Added inline comment marker/highlight logic

ssddanbrown committed:
Comments: Started inline comment display windows

MfeaR commented:
Hi! Could you please clarify if this feature will be implemented, and if so, when? We're evaluating documentation apps, and this is the only feature preventing us from choosing BookStack.

ssddanbrown commented:
@MfeaR This is a work in progress attempt at a feature. My aim is for this to be in the next feature release (so maybe late May or June) but can't promise anything.

ssddanbrown committed:
Comments: Styled content comments & improved interaction

ssddanbrown committed:
Comments: Added reference marker to comments

ssddanbrown committed:
Comments: Split out page comment reference logic to own component

Started support for editor view.
Moved comment elements to be added relative to content area instad of
specific target reference element.
Added relocating on screen size change.

ssddanbrown committed:
Comments: Added archive endpoints, messages, Js actions and tests

ssddanbrown committed:
Comments: Started archive display, created mode for tree node

ssddanbrown committed:
Comments: Moved to tab UI, Converted tabs component to ts

ssddanbrown committed:
Comments: Addressed a range of edge cases and ux issues for references

Handles only display and handling references when they're in the active
tab, while handling proper removal when made not visible.

ssddanbrown committed:
Comments: Further range of content reference ux improvements

- Added reference indicator to comment create form.
  - Added remove action.
- Extracted reference text to translations.
- Changed reference hash to be text-based instead of HTML based.
- Added reference display for newly added comments.
- Handled reference marker delete on comment delete.

ssddanbrown committed:
Comments: Fixed display, added archive list support for editor toolbox

ssddanbrown committed:
Comments: Checked content/arhived comment styles in dark mode

Also added default non-clickable styles for scenarios for references
which don't have an active content link.

ssddanbrown committed:
Comments: Split tests, added extra archive/reference tests

ssddanbrown committed:
Comments: Fixed a range of TS errors + other

- Migrated toolbox component to TS
- Aligned how custom event types are managed
- Fixed PHP use of content_ref where not provided

ssddanbrown committed:
Comments: Fixed pointer display, Fixed translation test

EDMJohnny commented:
Hi - thank you for implementing this feature.  One bug I've discovered is that after a comment is created, if you edit the comment, the reference link breaks, so the comment is no longer attached to the inline selection that started it.
